lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -877,6 +885,27 @@ getUTxOsStatistics ctx (ApiT wid) = do
   where
     liftE = throwE . ErrListUTxOStatisticsNoSuchWallet
 
+forceResyncWallet
+    :: forall ctx s t k.
+        ( ctx ~ ApiLayer s t k
+        )
+    => ctx
+    -> ApiT WalletId
+    -> Handler NoContent
+forceResyncWallet ctx (ApiT wid) = do
+    liftIO $ Registry.remove re wid
+    liftHandler (ExceptT safeRollback) `finally` liftIO (registerWorker ctx wid)
+    pure NoContent
+  where
+    re = ctx ^. workerRegistry @s @k
+    tr = ctx ^. logger
+    df = ctx ^. dbFactory @s @k
+    -- NOTE Safe because it happens without any worker running.
+    safeRollback = do
+        let tr' = Registry.transformTrace wid tr
```

Comment:
we will have log as coming from withLogging about request..I wonder if we do not want to say more here. although we will have also additional logs from `rollbackBlocks` below

Reply:
logs look as follows:

```
[cardano-wallet.api-server:Info:526] [2020-01-15 10:38:04.97 UTC] [RequestId 10] [DELETE] /v2/wallets/bdb212ccd1d556b62a59f4a4a1de0cff29f61ac1/utxos
[cardano-wallet.wallet-engine:Info:142] [2020-01-15 10:38:04.97 UTC] bdb212cc: Worker has exited: main action is over.
[cardano-wallet.wallet-engine:Info:526] [2020-01-15 10:38:04.98 UTC] bdb212cc: Try rolling back to 0.0
[cardano-wallet.wallet-engine:Info:526] [2020-01-15 10:38:05.00 UTC] bdb212cc: Rolled back to 0.0
[cardano-wallet.api-server:Info:526] [2020-01-15 10:38:05.01 UTC] [RequestId 10] 204 No Content in 0.044038928s
[cardano-wallet.wallet-engine:Info:528] [2020-01-15 10:38:05.03 UTC] bdb212cc: Applying blocks [1136063.1 ... 1144130.1]
[cardano-wallet.wallet-engine:Info:528] [2020-01-15 10:38:05.03 UTC] bdb212cc: Creating checkpoint at feb1b2c2-[1136063.1#1]
[cardano-wallet.wallet-engine:Info:528] [2020-01-15 10:38:05.04 UTC] bdb212cc: Creating checkpoint at 8920cbb4-[1136063.2#2]
[cardano-wallet.wallet-engine:Info:528] [2020-01-15 10:38:05.04 UTC] bdb212cc: Creating checkpoint at 73a21e9f-[1136063.3#3]
[...]
```

I think it's quite sufficient here :thinking:

Reply:
indeed, logs looks good. `Worker has exited...` logs removing of registry. and as expected there are logs from rollbackBlocks